slick/src/main/scala/slick/dbio/DBIOAction.scala

```diff
@@ -114,6 +114,22 @@ sealed trait DBIOAction[+R, +S <: NoStream, -E <: Effect] extends Dumpable {
   def collect[R2](pf: PartialFunction[R,R2])(implicit executor: ExecutionContext): DBIOAction[R2, NoStream, E] =
     map(r1 => pf.applyOrElse(r1,(r:R) => throw new NoSuchElementException(s"DBIOAction.collect partial function is not defined at: $r")))
 
+  /** Replace a result of a successful execution of this action with the unit value.
+    * If this action fails, the resulting action also fails.
+    *
+    * A shortcut for `.map(_ => ())`.
+    */
+  def void(implicit executor: ExecutionContext): DBIOAction[Unit, NoStream, E] =
+    map(_ => ())
+
+  /** Replace a result of a successful execution of this action with the given value.
+    * If this action fails, the resulting action also fails.
+    *
+    * A shortcut for `.map(_ => a)`.
+    */
+  def as[A](a: A)(implicit executor: ExecutionContext): DBIOAction[A, NoStream, E] =
```

Comment:
I think we can avoid taking an ExecutionContext by using the parasitic execution context.

Also, perhaps the parameter should be call-by-name. If previous actions fail it may never be needed. Also it would affect when the parameter would be evaluated. What signature do other libraries (e.g. cats-effect and ZIO) use?

Reply:
Cats-effect - eager - https://github.com/typelevel/cats-effect/blob/v3.5.4/core/shared/src/main/scala/cats/effect/IO.scala#L166

ZIO - call-by-name - https://github.com/zio/zio/blob/v2.0.21/core/shared/src/main/scala/zio/ZIO.scala#L187.

Reply:
I guess we go with call-by-name.